rust/bridge/shared/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2020 Signal Messenger, LLC.
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+#![allow(clippy::missing_safety_doc)]
+#![deny(warnings)]
+
+use aes_gcm_siv::Aes256GcmSiv;
+use libsignal_protocol_rust::*;
+use std::convert::TryFrom;
+
+#[cfg(not(any(feature = "ffi", feature = "jni")))]
+compile_error!("Either feature \"ffi\" or \"jni\" must be enabled for this crate.");
```

Comment:
Yeah, I went back and forth on this too, but without it you can't do things like `cargo clippy --all`, because `libsignal-ffi` and `libsignal-jni` depend on different features. It's not such a burden to support in the end, and I don't think we're likely to ship a combined lib by accident since the ways we build for clients always use scripts.